File: java/src/test/java/problems1_100/LengthOfLastWordTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LengthOfLastWordTest extends SampledTest {
    @Test
    public void test() {
        for (LengthOfLastWord solution : LengthOfLastWord.values()) {
            for (LengthOfLastWordSample sample : getSamples(LengthOfLastWordSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.lengthOfLastWord(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/LengthOfLastWordTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class LengthOfLastWordTest : SampledTest() {
    @Test
    fun test() =
        LengthOfLastWord.entries.forEach { solution ->
            getSamples(Array<LengthOfLastWordSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.lengthOfLastWord(input.s))
                        .isEqualTo(output)
                }
        }
}
